app/models/concepts_graph.rb

```diff
@@ -0,0 +1,98 @@
+# ConceptGraph transforms a track configuration to a form compatible for manipulating for
+# displaying the progression of the track
+class ConceptsGraph
```

Comment:
Mandate will give us some useful helper methods - not least `memoize`.

```suggestion
class ConceptsGraph
  include Mandate
```